pytest_base_url/plugin.py

```diff
@@ -21,13 +24,21 @@ def base_url(request):
 def _verify_url(request, base_url):
     """Verifies the base URL"""
     verify = request.config.option.verify_base_url
+    verify_timeout = request.config.getoption('verify_base_url_attempts')
     if base_url and verify:
-        response = requests.get(base_url, timeout=10)
-        if not response.status_code == requests.codes.ok:
-            raise pytest.UsageError(
-                'Base URL failed verification!'
-                '\nURL: {0}, Response status code: {1.status_code}'
-                '\nResponse headers: {1.headers}'.format(base_url, response))
+        session = requests.Session()
+        retries = Retry(total=verify_timeout,
+                        backoff_factor=0.1,
+                        status_forcelist=[500, 502, 503, 504])
+        session.mount(base_url, HTTPAdapter(max_retries=retries))
+        try:
+            session.get(base_url)
+        except Exception:
```

Comment:
The host is 'request.get'? I think you're right, this isn't a very useful exception. Let's keep ours, but it looks like we could distinguish between a bad status code using `retries.raise_on_status` even if we can't based on the exception type. Also, let's include the URL in the message as we did previously. The more information we can include, the more useful this will be.